ee/desktop/menu/menu_parse.go

```diff
@@ -0,0 +1,33 @@
+package menu
+
+func parseMenuData(m *MenuData, builder MenuBuilder) {
+	// Set top-level systray properties
+	builder.SetTooltip(m.Tooltip)
+
+	for _, child := range m.Items {
+		parseMenuItem(&child, builder, nil)
+	}
+}
+
+func parseMenuItem(m *MenuItemData, builder MenuBuilder, parent any) {
+	if m == nil {
+		return
+	}
+
+	if m.IsSeparator {
+		// If the item is a separator, nothing else matters
+		builder.AddSeparator()
+		return
+	}
+
+	var item any
+	if m.Label != "" {
+		// A menu item must have a non-empty label
+		item = builder.AddMenuItem(m.Label, m.Tooltip, m.Disabled, m.NonProdOnly, m.Action, parent)
+	}
+
+	for _, child := range m.Items {
+		// Recursively parse sub menu items
+		parseMenuItem(&child, builder, item)
```

Comment:
I'm a little weary of using recursion here, but it does the job. I can't imagine we would ever have more than 1 or 2 levels of sub menus, maybe it's worth unwinding this or putting a small limit on the recursion depth.

Reply:
I think recursion is correct. But it would also be reasonable to add a depth check.